AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreEndToEndTests.swift

```diff
@@ -513,40 +513,40 @@ class DataStoreEndToEndTests: SyncEngineIntegrationTestBase {
 
         let postsDeletedFromCloud = expectation(description: "All posts deleted and synced to cloud")
         postsDeletedFromCloud.expectedFulfillmentCount = count
-        var postsDeletedFromCloudCount = 0
         
-        sink = Amplify.DataStore.publisher(for: Post.self).sink { completed in
-            switch completed {
-            case .finished:
-                break
-            case .failure(let error):
-                XCTFail("\(error)")
-            }
-        } receiveValue: { mutationEvent in
-            guard posts.contains(where: { $0.id == mutationEvent.modelId }) else {
-                return
-            }
+        let task2 = Task {
+            var postsDeletedFromCloudCount = 0
+            let mutationEvents = Amplify.DataStore.observe(Post.self)
+            do {
+                for try await mutationEvent in mutationEvents {
+                    guard capturedPosts.contains(where: { $0.id == mutationEvent.modelId }) else {
+                        return
+                    }
 
-            if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
-                      mutationEvent.version == 1 {
-                self.log.debug(
-                    "Post deleted locally \(mutationEvent.modelId)")
-                postsDeletedLocally.fulfill()
-            } else if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
-                      mutationEvent.version == 2 {
-                postsDeletedFromCloudCount += 1
-                self.log.debug(
-                    "Post deleted and synced from cloud \(mutationEvent.modelId) \(postsDeletedFromCloudCount)")
-                postsDeletedFromCloud.fulfill()
+                    if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
+                              mutationEvent.version == 1 {
+                        self.log.debug(
+                            "Post deleted locally \(mutationEvent.modelId)")
+                        postsDeletedLocally.fulfill()
+                    } else if mutationEvent.mutationType == MutationEvent.MutationType.delete.rawValue,
+                              mutationEvent.version == 2 {
+                        postsDeletedFromCloudCount += 1
+                        self.log.debug(
+                            "Post deleted and synced from cloud \(mutationEvent.modelId) \(postsDeletedFromCloudCount)")
+                        postsDeletedFromCloud.fulfill()
+                    }
+                }
+            } catch {
+                XCTFail("Failed \(error)")
             }
         }
+        
         DispatchQueue.concurrentPerform(iterations: count) { index in
```

Comment:
This could be changed to use TaskGroup instead. That way it will stay in the Swift Concurrency thread pool. We can discuss after the PR is merged.